vendor/engines/secure_rooms/app/models/secure_rooms/occupancy.rb

```diff
@@ -0,0 +1,32 @@
+module SecureRooms
+
+  class Occupancy < ActiveRecord::Base
+
+    belongs_to :secure_room, foreign_key: :product_id
+    belongs_to :user
+    belongs_to :account
+    belongs_to :entry_event, class_name: SecureRooms::Event
+    belongs_to :exit_event, class_name: SecureRooms::Event
+
+    validates :product_id, :user_id, presence: true
+
+    def mark_oprhaned!
+      update!(orphan: Time.current)
+    end
+
+    def self.valid
+      where.not(orphan: nil)
+    end
+
+    def self.current
+      valid.where(exit_event_id: nil).where.not(entry_event_id: nil)
+    end
+
+    def self.recent
```

Comment:
Would `recent_exits` be a more specific name? Although, leaving it as `recent` so we can swap out 24 hours for recent 10 makes sense. I'm good either way.